doc/rtd/topics/code_review.rst

```diff
@@ -0,0 +1,170 @@
+   stale reviews, parallel reviews from multiple committers, etc.  If
+   we find that handling these on an ad-hoc basis is causing problems,
+   then we should revisit and expand this process to include them.
+
+   (If the **Proposer** is a core developer, then they are not
+   considered part of this group for the purposes of this process.)
+
+Opening Phase
+=============
+
+In this phase, the **Proposer** is responsible for all actions. They
+should get a PR into good enough shape that it is worth **the
+Committers** spending time reviewing it.  Specifically, they are
+responsible for getting to a point where the continuous integration
+(CI) testing in Travis is passing.  Once a PR is passing CI, it moves
+into the :ref:`Review phase`.
+
+These are the steps that comprise the opening phase:
+
+1. The **Proposer** opens PR
+
+2. An automated comment will be added to the PR, outlining the steps
+   expected of the **Proposer** before a **Committer** will review
+   their PR.  (N.B. This automation is not yet implemented!)
+
+3. CI runs automatically, and if
+
+   CI fails
+      The **Proposer** is expected to fix CI failures.  If the
+      **Proposer** doesn't understand the nature of the failures they
+      are seeing, they should comment in the pull request to request
+      assistance.  Alternatively, for more immediate assistance, they
+      can ask in ``#cloud-init`` on the Freenode IRC network.
+
+   CI passes
+      Move on to the :ref:`Review phase`.
+
+Review Phase
+============
+
+In this phase, the **Proposer** and the **Committers** will iterate
+together to, hopefully, get the PR merged into the cloud-init codebase.
+There are three potential outcomes: merged, rejected permanently, and
+temporarily closed.
+
+These are the steps that comprise the review phase:
+
+1. **The Committers** assign a **Committer** to the PR
+
+   The cloud-init core developers will work amongst themselves to
+   ensure that this happens in a timely fashion.  (Generally,
+   assignment will happen at the team's daily internal meeting.)  Note
+   that, in GitHub terms, this is setting an Assignee, not requesting a
+   review.
+
+2. That **Committer** reviews the PR, resulting in one of the
+   following:
+
+   Approve
+     The **Committer** will squash merge immediately.
+
+     There may be circumstances where a PR should not be merged
+     immediately; these will be handled case-by-case, and the
+     **Proposer** should make this requirement very clear in their pull
+     request description.
+
+     The review process is **DONE**.
+
+   Approve (with nits)
+     If the **Proposer** submits their PR with "Allow edits from
+     maintainer" enabled, and the only changes the **Committer**
+     requests are minor "nits", the **Committer** can push fixes for
+     those nits and squash merge.  (If a **Proposer** wants to opt-out
+     of this, then they should uncheck "Allow edits from maintainer"
+     when submitting their PR.)
```

Comment:
there's also the option to propose a change in github terms

Reply:
This feels like a tool that a reviewer can use, but I don't know that it really changes the code review _process_. The pull request will still either go via "Approve with nits" or "Request changes", "propose a change" is just a different way of expressing the nits and/or requested changes for those paths.

If you'd still like to see something added or change, could you propose a specific change that we can use as a starting point for discussion?

(Thanks for the review!)

Reply:
"Approve with nits" is basically a conditional approval. These IMHO are not helpful. The PR is either approved and can be merged or it needs changes. This of course does not resolve the cases where there are nits, which by definition are more or less things that can be debated forever without a real answer or satisfactory conclusion.

So if the reviewer has nits the reviewer should open a PR against the branch of the proposer and then the proposer can merge the work and the PR then moves forward. If the reviewer has nits but hey are not worth the reviewers time to submit a PR then they are not worth fixing and the PR should be merged. 

My $0.02

Reply:
OK, I'll need to reword this to state the overall outcome first, and then the details, because it specifically _isn't_ a conditional approval. This is the crucial part (bold added for this comment):

> the Committer can push fixes for those nits and **squash merge**.

"Approve (with nits)" is a committer saying, "this is an Approve except for these nits, let me fix those for you (to save a review round-trip) and merge".

Reading the current text again with fresh eyes, this could be a lot clearer.